poi/src/test/java/org/apache/poi/ss/util/BaseTestCellUtil.java

```diff
@@ -104,6 +105,126 @@ void setCellStylePropertyBorderWithShortAndEnum() throws IOException {
+            cs.setBorderTop(BorderStyle.THIN);
+            cs.setBottomBorderColor(IndexedColors.RED.index);
+            cs.setLeftBorderColor(IndexedColors.RED.index);
+            cs.setRightBorderColor(IndexedColors.RED.index);
+            cs.setTopBorderColor(IndexedColors.RED.index);
+            cs.setDataFormat(wb.createDataFormat().getFormat("#,##0"));
+            cs.setFillForegroundColor(IndexedColors.BLUE.index);
+            cs.setFillBackgroundColor(IndexedColors.BLUE.index);
+            cs.setFillPattern(FillPatternType.DIAMONDS);
+            cs.setFont(f);
+            cs.setHidden(true);
+            cs.setIndention((short) 3);
+            cs.setLocked(false);
+            cs.setRotation((short) 45);
+            cs.setVerticalAlignment(VerticalAlignment.CENTER);
+            cs.setWrapText(true);
+            cs.setShrinkToFit(true);
+            cs.setQuotePrefixed(true);
+            c.setCellStyle(cs);
+
+            // Set BorderBottom from THIN to DOUBLE with setCellStyleProperty()
+            CellUtil.setCellStyleProperty(c, CellUtil.BORDER_BOTTOM, BorderStyle.DOUBLE);
+
+            // Assert that only BorderBottom has been changed and no others.
+            assertEquals(BorderStyle.DOUBLE, c.getCellStyle().getBorderBottom());
+            assertEquals(HorizontalAlignment.CENTER, c.getCellStyle().getAlignment());
+            assertEquals(BorderStyle.THIN, c.getCellStyle().getBorderLeft());
+            assertEquals(BorderStyle.THIN, c.getCellStyle().getBorderRight());
+            assertEquals(BorderStyle.THIN, c.getCellStyle().getBorderTop());
+            assertEquals(IndexedColors.RED.index, c.getCellStyle().getBottomBorderColor());
+            assertEquals(IndexedColors.RED.index, c.getCellStyle().getLeftBorderColor());
+            assertEquals(IndexedColors.RED.index, c.getCellStyle().getRightBorderColor());
+            assertEquals(IndexedColors.RED.index, c.getCellStyle().getTopBorderColor());
+            assertEquals(wb.createDataFormat().getFormat("#,##0"), c.getCellStyle().getDataFormat());
+            assertEquals(IndexedColors.BLUE.index, c.getCellStyle().getFillForegroundColor());
+            assertEquals(IndexedColors.BLUE.index, c.getCellStyle().getFillBackgroundColor());
+            assertEquals(FillPatternType.DIAMONDS, c.getCellStyle().getFillPattern());
+            assertEquals(f.getIndex(), c.getCellStyle().getFontIndex());
+            assertTrue(c.getCellStyle().getHidden());
+            assertEquals((short) 3, c.getCellStyle().getIndention());
+            assertFalse(c.getCellStyle().getLocked());
+            assertEquals((short) 45, c.getCellStyle().getRotation());
+            assertEquals(VerticalAlignment.CENTER, c.getCellStyle().getVerticalAlignment());
+            assertTrue(c.getCellStyle().getWrapText());
+            assertTrue(c.getCellStyle().getShrinkToFit());
+            assertTrue(c.getCellStyle().getQuotePrefixed());
```

Comment:
Before the fix, `c.getCellStyle().getShrinkToFit()` would be `false` here and the test would fail.